packages/vite/src/node/optimizer/index.ts

```diff
@@ -290,7 +299,11 @@ function needsInterop(
   const flatId = flattenId(id) + '.js'
   let generatedExports: string[] | undefined
   for (const output in outputs) {
-    if (normalizePath(output).endsWith('.vite/' + flatId)) {
+    if (
+      normalizePath(output).endsWith(
+        normalizePath(path.join(cacheDirOutputPath, flatId))
```

Comment:
This check could generate issues in the feature, no? `nodes_modules/.vite` is quite unique but if the user ends up selecting something like `cache`, there could be a collision. Could we check for the cache dir with startsWidth from the root? Or was this using endsWith for another reason?

Reply:
In fact, we can use `===` here. But I need to confirm if it would work properly on Windows